OptimizelySDK.Tests/OptimizelyConfigTests/OptimizelyConfigTest.cs

```diff
@@ -326,7 +397,61 @@ public void TestGetOptimizelyConfigService()
+            OptimizelyConfig expectedOptimizelyConfig = new OptimizelyConfig(datafileProjectConfig.Revision,
+                datafileProjectConfig.SDKKey,
+                datafileProjectConfig.EnvironmentKey,
+                attributes: new Entity.Attribute[]
+                {
+                    new Entity.Attribute
+                    {
+                       Id = "594015", Key = "house"
+                    },
+                    new Entity.Attribute
+                    {
+                       Id = "594016", Key = "lasers"
+                    },
+                    new Entity.Attribute
+                    {
+                       Id = "594017", Key = "should_do_it"
+                    },
+                    new Entity.Attribute
+                    {
+                       Id = "594018", Key = "favorite_ice_cream"
+                    }
+                },
+                audiences: new OptimizelyAudience[]
+                {
+                    new OptimizelyAudience("3468206642", "exactString", "[\"and\", [\"or\", [\"or\", {\"name\": \"house\", \"type\": \"custom_attribute\", \"value\": \"Gryffindor\"}]]]"),
+                    new OptimizelyAudience("3468206645", "notChrome", "[\"and\", [\"or\", [\"not\", [\"or\", {\"name\": \"browser_type\", \"type\": \"custom_attribute\", \"value\":\"Chrome\"}]]]]"),
+                    new OptimizelyAudience("3988293898", "$$dummySubstringString", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3988293899", "$$dummyExists", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3468206646", "$$dummyExactNumber", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3468206647", "$$dummyGtNumber", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3468206644", "$$dummyLtNumber", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3468206643", "$$dummyExactBoolean", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("0", "$$dummy", "{\"type\": \"custom_attribute\", \"name\": \"$opt_dummy_attribute\", \"value\": \"impossible_value\"}"),
+                    new OptimizelyAudience("3988293898", "substringString", new object[] { "and", new object[] { "or", new object[] { "or", new Dictionary<string, string>() { { "name", "house" }, { "type", "custom_attribute" }, { "match", "substring" }, { "value", "Slytherin" } } } } }),
```

Comment:
Can we modify tests to validate that typedAudiences and audiences filter out same "names"?